### PR TITLE
Update NPCs.md

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary

SUMMARY: Content "Add NPC_class and NPC json templates and instructions"

#### Purpose of change

I saw there was a todo to demonstrate how to create new NPCs in the documentation and decided to take a stab at it.  I'd love for people to make suggestions where needed as this is my largest documentation work I've done so far.

### CODE REVIEW

File: doc/NPCs.md [hunk 1]
```diff
@@ -1,4 +1,54 @@
...
+  "name_suffix": "examplar",
+  "class": "NC_EXAMPLE",
+  "attitude": 0,
+  "mission": 7,
+  "chat": "TALK_EXAMPLE",
+  "faction": "no_faction"
+}
+```
+This is the json that creates the NPC ID that is used to spawn an NPC in mapgen.
+Attitude is based on the enum in npc.h. The important ones are 0=NPCATT_NULL, 1=NPCATT_TALK", 3=NPCATT_FOLLOW, 7=NPCATT_DEFEND, 10=NPCATT_KILL, and 11=NPCATT_FLEE.
```
Comment (feinorgh) [inline doc/NPCs.md:48]:
```suggestion
Attitude is based on the enum in [src/npc.h]. The important ones are:
* [NPCATT_NULL](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L87) (0)
* [NPCATT_TALK](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L88) (1)
* [NPCATT_FOLLOW](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L90) (3)
* NPCATT_DEFEND (7)
* [NPCATT_KILL](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L97) (10)
* [NPCATT_FLEE](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L98) (11)

File: doc/NPCs.md [hunk 2]
```diff
@@ -1,4 +1,54 @@
...
+Mission is based on the enum in npc.h.  The important ones are 0=NPC_MISSION_NUL, 3=NPC_MISSION_SHOPKEEP", and 7=NPC_MISSION_GUARD", 8 = NPC_MISSION_GUARD_PATROL will actively investigate noises".
```
Comment (feinorgh) [inline doc/NPCs.md:49]:
```suggestion
Mission is based on the enum in [src/npc.h].  The important ones are
* [NPC_MISSION_NUL](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L185) (0)
* [NPC_MISSION_SHOPKEEP](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L188) (3)
* [NPC_MISSION_GUARD](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L194) (7)
* [NPC_MISSION_GUARD_PATROL](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L195) (8) will actively investigate noises".
```

File: doc/NPCs.md [hunk 3]
```diff
@@ -1,4 +1,54 @@
...
+Chat is covered in the dialogue examples below.
+Faction determines what faction, if any, the NPC belongs to.  Some examples are the Free Traders, Old Guard, Marloss Evangelists and Hell's raiders but could include a brand new faction you create!
```
Comment (feinorgh) [inline doc/NPCs.md:51]:
```suggestion
Faction determines what faction, if any, the NPC belongs to.  Some examples are the Free Traders, Old Guard, Marloss Evangelists, and Hell's raiders but could include a brand new faction you create!
```

File: doc/NPCs.md [hunk 4]
```diff
@@ -1,4 +1,54 @@
...
 
 # Writing dialogues
 Dialogues work like state machines. They start with a certain topic (the NPC says something), the player character can respond (choosing one of several responses), and that response sets the new talk topic. This goes on until the dialogue is finished, or the NPC turns hostile.
```
Comment (feinorgh) [inline doc/NPCs.md:54]:
```suggestion
Dialogues work like state machines. They start with a certain topic (the NPC says something), the player character can then respond (choosing one of several responses), and that response sets the new talk topic. This goes on until the dialogue is finished, or the NPC turns hostile.
```

File: doc/NPCs.md (hunk 1 above)
Comment (Maleclypse) [reply doc/NPCs.md:48]: The original text is from the notes for an NPC in the refugee center.

File: doc/NPCs.md (hunk 2 above)
Comment (Maleclypse) [reply doc/NPCs.md:49]: The original text is from the notes for an NPC in the refugee center.

**Pull request merged**